composer.json

```diff
@@ -24,23 +24,24 @@
         "webmozart/assert": "^1.2"
     },
     "require-dev": {
+        "humbug/box": "^3.0@dev",
```

Comment:
If you are using `bamarni/composer-bin-plugin` I would recommend you to install `humbug/box` with it as the later doesn't need to execute any of your code

Reply:
How would you automate that to don't bother user with that?
I've added it to `composer.json` script, but it takes too much perfromance.

Reply:
you could do `composer bin {box|php-scoper} install` in your composer script. Even if unnecessary once already installed it's like not even half a second. Alternatively I'm using Makefiles lately for this kind of stuff. You can see some examples on alice, AliceDataFixture, php-scoper or box: just do `make test` or `make build` and everything is taken care of

Reply:
That would still bother the user.

Maybe add bash script, that would check if vendor-bin directory for package exist, and install it only if not. What do you think?

Reply:
Yeah you can definitely have a `compile` bash script or something instead.

Remember most of your users/contributors won't have to build the PHAR so I don't think it's gonna bother them much. But otherwise a bash/Makefile gives more flexibility on how to do things

Reply:
I'll look on it.

> Remember most of your users/contributors won't have to build the PHAR so I don't think it's gonna bother them much. But otherwise a bash/Makefile gives more flexibility on how to do things

Speed matter for us as well.

Reply:
Yeah but I think the time taken by composer to dump the autoloader (which is the only thing it will do if the packages are already installed) is negligible compared to the rest.

IMO the places of this process where perf. can be improved:

- The scoping itself, i.e. the core of PHP-Scoper
- PHP-Scoper as standalone as we could leverage parallelism for example. But I don't think I'll invest much time on it tbh. I'm more interesting in integrating with Box which case the parallelisation already happens there
- Box itself
- PHP-Parser

Any optimisation in any of those above is more likely to have a higher impact that the half second spent on dumping the autoloader.

That said if it's bugging you, there's nothing wrong with using bash/Makefile to handle this